src/coreclr/inc/winwrap.h

```diff
@@ -254,18 +254,32 @@ WszCreateProcess(
     LPPROCESS_INFORMATION lpProcessInformation
     );
 
-#if defined(HOST_X86) && defined(_MSC_VER)
+#if defined(_MSC_VER)
 
 //
-// Windows SDK does not use intrinsics on x86. Redefine the interlocked operations to use intrinsics.
+// Redefine the interlocked operations to intrinsics
+// to avoid missing InterlockedCompareExchange8 and Windows 8 requirement.
 //
 
 #include "intrin.h"
 
-#define InterlockedIncrement            _InterlockedIncrement
-#define InterlockedDecrement            _InterlockedDecrement
+#define InterlockedExchange8            _InterlockedExchange8
+#define InterlockedCompareExchange8     _InterlockedCompareExchange8
```

Comment:
If this is working around missing InterlockedCompareExchange8, the workaround should just have InterlockedCompareExchange8 - the rest is unnecessary.

Do you happen to know if there is a bug opened on InterlockedCompareExchange8 missing? If not, we should open one.

Reply:
> Do you happen to know if there is a bug opened on InterlockedCompareExchange8 missing? If not, we should open one.

Haven't checked if there's one since I don't really know where WinAPI issues go, this should probably be fixed there too but using it then would still not be safe since it'd only be guaranteed to be present in the future OS version DLLs.

Reply:
> Haven't checked if there's one since I don't really know where WinAPI issues go

https://github.com/microsoft/WindowsAppSDK seems to be the place to report Windows SDK issues on github.

Reply:
Opened https://github.com/microsoft/WindowsAppSDK/issues/4074